### PR TITLE
Added unittest for deployment

### LINKED ISSUE #17: Write unit tests for deployment part
Improve test coverage, by writing unit test for getting configmap based on a mock k8s object that can be generated like the one in https://github.com/devopsspiral/KubeLibrary/blob/d102ac0cc5caf573222053356d97f137862f04e0/test/test_KubeLibrary.py#L146 and store it in test/resources/. 

* Create test for generating object .json
* Write test that checks target keyword
* After coverage is up modify the CI threshold in .coveragerc to new value:
```
(...)
[report]
fail_under = 67
```

### PR DESCRIPTION
**Pull request by satish-nubolab**
Added unittest for deployments.

Fixes #17

### CODE REVIEW

**Comment by Nilsty** [comment]
Looks good to me. Please update docs/index.html for the new filter keyword.

**Comment by satish-nubolab** [comment]
Updated index.html.

File: src/KubeLibrary/KubeLibrary.py [hunk 1]
```diff
@@ -268,6 +268,14 @@ def get_secrets_in_namespace(self, name_pattern, namespace, label_selector=""):
         secrets = [item for item in ret.items if r.match(item.metadata.name)]
         return secrets
 
+    def filter_deployment_names(self, deployment):
```
Comment (m-wcislo) [inline src/KubeLibrary/KubeLibrary.py:271]: Small thing, should be deployments.

File: test/test_KubeLibrary.py [hunk 2]
```diff
@@ -39,11 +39,11 @@ def __getitem__(self, key):
         return getattr(self, key)
 
 
-def mock_list_namespaced_config_map(namespace, watch=False, label_selector=""):
```
Comment (m-wcislo) [inline test/test_KubeLibrary.py:42]: I believe you are removing a test here

File: test/test_KubeLibrary.py [hunk 3]
```diff
@@ -334,12 +334,12 @@ def test_get_role_bindings_in_namespace(self, mock_lnp):
         role_bindings = kl.get_role_bindings_in_namespace('default')
         self.assertEqual(['read-pods'], [item for item in role_bindings])
 
-    @mock.patch('kubernetes.client.CoreV1Api.list_namespaced_config_map')
```
Comment (m-wcislo) [inline test/test_KubeLibrary.py:337]: Same here, replace test instead of adding

**Review by m-wcislo — CHANGES_REQUESTED** [review]

File: src/KubeLibrary/KubeLibrary.py (hunk 1 above)
Comment (satish-nubolab) [reply src/KubeLibrary/KubeLibrary.py]: Done.

File: test/test_KubeLibrary.py (hunk 2 above)
Comment (satish-nubolab) [reply test/test_KubeLibrary.py]: Changes Done.

File: test/test_KubeLibrary.py (hunk 3 above)
Comment (satish-nubolab) [reply test/test_KubeLibrary.py:337]: Changes Done.

File: .coveragerc [hunk 4]
```diff
@@ -4,3 +4,4 @@ source =
     src/
 [report]
 fail_under = 72
```
Comment (m-wcislo) [inline .coveragerc:6]: ok resolve conflicts and update coverage and we can merge it

**Review by m-wcislo — CHANGES_REQUESTED** [review]

File: .coveragerc (hunk 4 above)
Comment (satish-nubolab) [reply .coveragerc]: Done

**Review by m-wcislo — APPROVED** [review]

**Pull request merged**